bundles/org.openhab.io.homekit/src/main/java/org/openhab/io/homekit/internal/HomekitImpl.java

```diff
@@ -211,60 +227,75 @@ private HomekitRoot startBridge(HomekitServer homekitServer, HomekitAuthInfoImpl
             }
+            logger.trace("create HomeKit server with dedicated mDNS server");
+            homekitServer = new HomekitServer(localNetworkInterface, settings.port + instance);
+        }
+        homekitServers.add(homekitServer);
+        HomekitChangeListener changeListener = new HomekitChangeListener(itemRegistry, settings, metadataRegistry,
+                storage, instance + 1);
+        changeListeners.add(changeListener);
+        startBridge(homekitServer, authInfo, changeListener, instance + 1);
+        authInfos.add(authInfo);
+    }
 
+    private void startHomekitServer() throws IOException, InvalidAlgorithmParameterException {
+        if (homekitServers.isEmpty()) {
             for (int i = 0; i < settings.instances; ++i) {
-                String storage_key = HomekitAuthInfoImpl.STORAGE_KEY;
-                if (i != 0) {
-                    storage_key += i;
-                }
-                Storage<Object> storage = storageService.getStorage(storage_key);
-                HomekitAuthInfoImpl authInfo = new HomekitAuthInfoImpl(storage, settings.pin, settings.setupId,
-                        settings.blockUserDeletion);
-
-                @Nullable
-                HomekitServer homekitServer = null;
-                if (settings.useOHmDNS) {
-                    for (JmDNS mdns : mdnsClient.getClientInstances()) {
-                        if (mdns.getInetAddress().equals(networkInterface)) {
-                            logger.trace("suitable mDNS client for IP {} found and will be used for HomeKit",
-                                    networkInterface);
-                            homekitServer = new HomekitServer(mdns, settings.port + i);
-                        }
-                    }
-                }
-                if (homekitServer == null) {
-                    if (settings.useOHmDNS) {
-                        logger.trace("no suitable mDNS server for IP {} found", networkInterface);
-                    }
-                    logger.trace("create HomeKit server with dedicated mDNS server");
-                    homekitServer = new HomekitServer(networkInterface, settings.port + i);
-                }
-                homekitServers.add(homekitServer);
-                HomekitChangeListener changeListener = new HomekitChangeListener(itemRegistry, settings,
-                        metadataRegistry, storage, i + 1);
-                changeListeners.add(changeListener);
-                bridges.add(startBridge(homekitServer, authInfo, changeListener, i + 1));
-                authInfos.add(authInfo);
+                startHomekitServer(i);
             }
         } else {
             logger.warn("trying to start HomeKit server but it is already initialized");
         }
     }
 
+    private InetAddress ensureNetworkInterface() throws IOException {
+        InetAddress localNetworkInterface = networkInterface;
```

Comment:
very minor thing but in case you will touch the code anyway 
i think we dont need to create localNetworkInterface here but just

if (networkInterface != null) { return networkInterface }